resources/fetching/fetchers/gcp_assets_fetcher.go

```diff
@@ -0,0 +1,115 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package fetchers
+
+import (
+	"context"
+	"fmt"
+
+	"cloud.google.com/go/asset/apiv1/assetpb"
+	"github.com/elastic/cloudbeat/resources/fetching"
+	"github.com/elastic/cloudbeat/resources/providers/gcplib/assets_inventory"
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+type GcpAssetsFetcher struct {
+	log        *logp.Logger
+	resourceCh chan fetching.ResourceInfo
+	provider   *assets_inventory.Provider
+}
+
+type GcpAsset struct {
+	Type    string
+	SubType string
+
+	Asset *assetpb.Asset `json:"asset,omitempty"`
+}
+
+func NewGcpAssetsFetcher(ctx context.Context, log *logp.Logger, ch chan fetching.ResourceInfo, provider *assets_inventory.Provider) *GcpAssetsFetcher {
+	return &GcpAssetsFetcher{
+		log:        log,
+		resourceCh: ch,
+		provider:   provider,
+	}
+}
+
+var gcpAssetTypes = map[string]map[string][]string{
+	fetching.KeyManagement: {
+		"gcp-kms": {"cloudkms.googleapis.com/CryptoKey"},
+	},
+	fetching.CloudIdentity: {
+		"gcp-iam": {"iam.googleapis.com/ServiceAccount"},
+	},
+	fetching.CloudDatabase: {
+		"gcp-bigquery": {"bigquery.googleapis.com/Dataset"},
```

Comment:
Let's add a suffix for this sub-type as we'll need to add `gcp-bigquery-table` for rule 7.2.
```suggestion
		"gcp-bigquery-dataset": {"bigquery.googleapis.com/Dataset"},
```